qiskit/utils/deprecation.py

```diff
@@ -353,9 +354,11 @@ def _write_deprecation_msg(
     pending: bool,
     additional_msg: str,
     removal_timeline: str,
-) -> Tuple[str, Union[Type[DeprecationWarning], Type[PendingDeprecationWarning]]]:
+) -> tuple[str, Type[DeprecationWarning] | Type[PendingDeprecationWarning]]:
     if pending:
-        category = PendingDeprecationWarning
+        category: Type[DeprecationWarning] | Type[
```

Comment:
I think it would be clearer to declare the type for `category` before the `if pending` line.